runner/runners/invoke.go

```diff
@@ -283,98 +273,105 @@ func (inv *Invoker) run(cmd *runner.Command, id runner.RunID, abortCh chan struc
 
 	switch st.State {
 	case execer.COMPLETE:
-		tmp, err := inv.tmp.TempDir("invoke")
-		if err != nil {
-			return runner.FailedStatus(id, fmt.Errorf("error staging ingestion dir: %v", err),
-				tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
-		}
-		uploadTimer := inv.stat.Latency(stats.WorkerUploadLatency_ms).Time()
-		inv.stat.Counter(stats.WorkerUploads).Inc(1)
-		defer func() {
-			os.RemoveAll(tmp.Dir)
-			uploadTimer.Stop()
-		}()
-		outPath := stdout.AsFile()
-		errPath := stderr.AsFile()
-		logPath := stdlog.AsFile()
-		stdoutName := "STDOUT"
-		stderrName := "STDERR"
-		stdlogName := "STDLOG"
-		var writer *os.File
-		var reader *os.File
-		defer writer.Close()
-		defer reader.Close()
-
-		if writer, err = os.Create(filepath.Join(tmp.Dir, stdoutName)); err != nil {
-			return runner.FailedStatus(id, fmt.Errorf("error staging ingestion for stdout: %v", err),
-				tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
-		} else if reader, err = os.Open(outPath); err != nil {
-			return runner.FailedStatus(id, fmt.Errorf("error staging ingestion for stdout: %v", err),
-				tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
-		} else if _, err := io.Copy(writer, reader); err != nil {
-			return runner.FailedStatus(id, fmt.Errorf("error staging ingestion for stdout: %v", err),
-				tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
-		}
+		if runType == runner.RunTypeScoot {
+			tmp, err := inv.tmp.TempDir("invoke")
```

Comment:
this is a lot of code in this if block - can it be pulled into a well-named/documented function?

Reply:
Had thought about that. As-is I left the "Scoot" type behavior inline and broke out the Bazel processing. Will break some of this out into a post-process piece